src/main/java/org/openpnp/machine/reference/feeder/Bank.java

```diff
@@ -0,0 +1,271 @@
+            @Override
+            public void configurationLoaded(Configuration configuration) throws Exception {
+		commit();
+            }
+        });
+    }
+
+
+    @Override
+    public void setEnabled(boolean enabled) {
+	ReferenceMachine machine = (ReferenceMachine)Configuration.get().getMachine();
+	String id=getId() + ":"; int i=-1;
+	if(this.enabled!=enabled) {
+		//mapLoc(this.enabled=enabled); 	// this stage
+	  	for(Feeder fdr:feeders) 	// deeper stages
+		if(fdr.getClass()==this.getClass())  // this class
+		if(this.enabled) ((Bank)fdr).commit(); else {
+			enabled=((Bank)fdr).enabled;
+			fdr.setEnabled(false);
+			((Bank)fdr).enabled=enabled;
+		}
+	}
+	if(!enabled) {
+	  for(Feeder fdr:feeders) {
+		machine.removeFeeder(fdr);
+	  }
+	  for(Feeder fdr:machine.getFeeders()) 
+		if(fdr.getName().startsWith(id))
+		machine.removeFeeder(fdr);
+	}
+
+	if( enabled) 
+	for(Feeder fdr:feeders) {
+			if(machine.getFeeder(fdr.getId())!=fdr) 
+			try { machine.addFeeder(fdr); } catch(Exception e) {;}
+	}
+    }
+
+
+    @Override
+    public void feed(Nozzle nozzle) throws Exception {
+	if(getId().length()==0) { // hack
+		setEnabled(true);
+		feeders.clear();
+		for(Board b:Configuration.get().getBoards()) 
+		for(Placement p:b.getPlacements()) 
+	  	for(Feeder fdr:machine.getFeeders()) 
+		if(!fdr.getPart().toString().equals(p.getPart().toString()))
+			feeders.add(fdr);		
+		setEnabled(false);
+		return;
+        }
+	// vars
+	if(nozzle==null) nozzle=machine.getDefaultHead().getDefaultNozzle();
+	Camera camera = nozzle.getHead().getDefaultCamera();
+	double mmx=camera.getUnitsPerPixel().getX();
+	double mmy=camera.getUnitsPerPixel().getY();
+	boolean ena = enabled;
+	// move camera
+	nozzle.moveToSafeZ();
+	camera.moveTo(location.derive(null,null,Double.NaN,0.0));
+	camera.moveToSafeZ();
+	// updte feeders
+	setEnabled(false);	
+	// changed from RTAG to barcode and to bufferedImage
+        CameraView cameraView = MainFrame.get().getCameraViews().setSelectedCamera(camera);
+	camera.settleAndCapture(); 
+	cameraView.setSelectionEnabled(true);
+	cameraView.setSelection(0,0,(int)Math.floor(mmx*cam_x+0.5),(int)Math.floor(mmy*cam_y+0.5));
+        String ids = new MultiFormatReader().decode(
+   	 		new BinaryBitmap(new HybridBinarizer(
+        			new BufferedImageLuminanceSource( 
+					cameraView.captureSelectionImage()
+		)))).getText();
+	cameraView.setSelectionEnabled(false);
+	// id now contains value
+	Feeder feeder=null; int n=0;
+	if(ids!=null)
+	for(String id:ids.split("|")) { 
+		logger.debug("found feeder {} at bank {}",id,getId());
+		for(Feeder fdr:machine.getFeeders())  
+			if(id.equals(fdr.getId())) { feeder=fdr; break; }
+		if(feeder!=null) {
+			logger.info("found feeder {} at bank {}",feeder.getName(),getName());
+			// register feeder
+		} else { String dir = System.getProperty("user.dir");
+			System.setProperty("user.dir",  new File(Configuration.get().getConfigurationDirectory(), "script").toString());
```

Comment:
Please do not modify user.dir. This can cause problems with the entire program. If your i.eval(id) fails, for instance, the old directory will not get set back correctly.